### PR TITLE
Add several North Carolina deduction variables

### LINKED ISSUE #2923: Add North Carolina deduction variables, parameters, and unit tests
Need to make these additions in order to make progress towards being able to start integration testing in issue #993.

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #2923.

When preparing this pull request, I did not see PR #2711 (wrt NC child deduction) or PR #2481 (wrt NC standard and itemized deductions).  Sorry about that, but when so many PRs are open, it is hard to find old pending PRs when they don't have any labels.

This PR makes PR #2711 redundant.
This PR makes (the non-itemized) part of PR #2481 redundant.

So, if this PR was merged, PR #2711 could be closed without merging,
and PR #2481 could be simplified to include just a `nc_itemized_deductions` variable, parameters, and unit tests.

### Summary
:sparkles::memo::white_check_mark:

This pull request adds several variables and parameters for the North Carolina income tax deductions, as well as unit tests and a changelog entry. It implements the logic for the child deduction and the standard deduction, and creates a placeholder for the itemized deduction. It also adds a variable that returns the larger of the standard or itemized deductions. These changes enable the policy engine to calculate the North Carolina income tax liability more accurately and allow integration testing.

> _`North Carolina`_
> _Deductions added to engine_
> _`Child`, `standard`, `itemized`_

### Walkthrough
*  Add North Carolina deduction variables and tests ([link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8L1-R3), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-529c2643bf902023f37e7c669c211f18b01a95e39dbf308e9f979d666df322d2R1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-43df7bae5634c54e9de0db8e0f92df06296b65ea3ebd5b7b6e0b59ca2706388cR1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-8a7b3ed258ce48e577fa7742088aabd1b57245f66474e94c14f948ce7cbc7b13R1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-177e0cb75aab7182237608eda95504d18ccdb995ef8a97dbd3142364e02b765bR1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-40ef5c377c36ae75e6f628684ebf4d40f24b896c33078abb67a6b82a60aaad17R1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-dd5dbef27b43fca869e916be1bfe6c930c4ed9223cf47cb85afe8c9defdccb37R1-R28), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-4f17f882e34acea79f93e09408380509509dac7bedbe7da4387dc95c338dd9cbR1-R59), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-6f9bfe316c422f513545f87c7fcb68538fdf530936a48117aa0c59bacfaf76e8R1-R23), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-2335e41ac9126f6042b6bd5e316f49071f752ef65eaaa11d525e67a5b0dabfa5R1-R8), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-78b28dc8da8fb258317b2e5eb280d65aac7cccbb8104995693004e17efab23ceR1-R37), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-e0aac7f1eb72838c94dc76b31f98a760e6d00a9bb851f8f7ce315e4670dac6bdR1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-2a49278c26ceed853578ec9dba925d131aba64f248da78d6e9802f8e3e4499deR1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-05462a95c07df18c0f2f4a068c465a8e7ec53047b5bb40a2bad5787b313e394eR1-R25))
  * Create parameter files for the child deduction per eligible child by filing status (`child/head_of_household.yaml`, `child/joint.yaml`, `child/separate.yaml`, `child/single.yaml`, `child/widow.yaml`) and the standard deduction amount by filing status (`standard/amount.yaml`) with descriptions, metadata, brackets, and references ([link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-529c2643bf902023f37e7c669c211f18b01a95e39dbf308e9f979d666df322d2R1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-43df7bae5634c54e9de0db8e0f92df06296b65ea3ebd5b7b6e0b59ca2706388cR1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-8a7b3ed258ce48e577fa7742088aabd1b57245f66474e94c14f948ce7cbc7b13R1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-177e0cb75aab7182237608eda95504d18ccdb995ef8a97dbd3142364e02b765bR1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-40ef5c377c36ae75e6f628684ebf4d40f24b896c33078abb67a6b82a60aaad17R1-R48), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-dd5dbef27b43fca869e916be1bfe6c930c4ed9223cf47cb85afe8c9defdccb37R1-R28))
  * Define variable files for the child deduction (`nc_child_deduction.py`), the itemized deductions (`nc_itemized_deductions.py`), the standard deduction (`nc_standard_deduction.py`), and the standard or itemized deductions (`nc_standard_or_itemized_deductions.py`) with value types, entities, labels, units, definition periods, formulas, and references ([link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-78b28dc8da8fb258317b2e5eb280d65aac7cccbb8104995693004e17efab23ceR1-R37), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-e0aac7f1eb72838c94dc76b31f98a760e6d00a9bb851f8f7ce315e4670dac6bdR1-R10), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-2a49278c26ceed853578ec9dba925d131aba64f248da78d6e9802f8e3e4499deR1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-05462a95c07df18c0f2f4a068c465a8e7ec53047b5bb40a2bad5787b313e394eR1-R25))
  * Implement formulas for the child deduction, the standard deduction, and the standard or itemized deductions that use the parameters and other variables ([link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-78b28dc8da8fb258317b2e5eb280d65aac7cccbb8104995693004e17efab23ceR1-R37), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-2a49278c26ceed853578ec9dba925d131aba64f248da78d6e9802f8e3e4499deR1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-05462a95c07df18c0f2f4a068c465a8e7ec53047b5bb40a2bad5787b313e394eR1-R25))
  * Add unit test files for the child deduction (`nc_child_deduction.yaml`), the standard deduction (`nc_standard_deduction.yaml`), and the standard or itemized deductions (`nc_standard_or_itemized_deductions.yaml`) with test cases that cover different scenarios and check the expected outputs ([link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-4f17f882e34acea79f93e09408380509509dac7bedbe7da4387dc95c338dd9cbR1-R59), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-6f9bfe316c422f513545f87c7fcb68538fdf530936a48117aa0c59bacfaf76e8R1-R23), [link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-2335e41ac9126f6042b6bd5e316f49071f752ef65eaaa11d525e67a5b0dabfa5R1-R8))
  * Update the changelog entry to indicate the addition of the North Carolina deduction variables and the minor version bump ([link](https://github.com/PolicyEngine/policyengine-us/pull/2924/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8L1-R3))

### CODE REVIEW

**Comment by PavelMakarchuk** [comment]
Would this close PR #2711 and #2481 ?

**Comment by martinholmer** [comment]
@PavelMakarchuk,  See the edited lead comment in the discussion of PR #2924 regarding your questions.

File: policyengine_us/parameters/gov/states/nc/tax/income/deductions/child/head_of_household.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+description: North Carolina provides a child deduction of this amount per eligible child for head-of-household filers.
+
+metadata:
+  label: North Carolina child deduction per eligible child for head-of-household filers.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/income/deductions/child/head_of_household.yaml:4]:
```suggestion
  label: North Carolina child deduction per eligible child for head-of-household filers
```

File: policyengine_us/parameters/gov/states/nc/tax/income/deductions/child/joint.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+description: North Carolina provides a child deduction of this amount per eligible child for joint filers.
+
+metadata:
+  label: North Carolina child deduction per eligible child for joint filers.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/income/deductions/child/joint.yaml:4]:
```suggestion
  label: North Carolina child deduction per eligible child for joint filers
```

File: policyengine_us/parameters/gov/states/nc/tax/income/deductions/child/separate.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+description: North Carolina provides a child deduction of this amount per eligible child for separate filers.
+
+metadata:
+  label: North Carolina child deduction per eligible child for separate filers.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/income/deductions/child/separate.yaml:4]:
```suggestion
  label: North Carolina child deduction per eligible child for separate filers
```

File: policyengine_us/parameters/gov/states/nc/tax/income/deductions/child/single.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+description: North Carolina provides a child deduction of this amount per eligible child for single filers.
+
+metadata:
+  label: North Carolina child deduction per eligible child for single filers.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/income/deductions/child/single.yaml:4]:
```suggestion
  label: North Carolina child deduction per eligible child for single filers
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**